autoload/utils/cmake.vim

```diff
@@ -88,16 +89,23 @@ function! s:populateDefaultCMakeVariants() abort
     endif
 endfunction
 " }}} Private functions "
+"
+function! utils#cmake#getLoadedCMakeKits() abort
+    if empty(s:loaded_cmake_kits)
+        return utils#cmake#kits#getCMakeKits()
+    endif
+    return s:loaded_cmake_kits
+endfunction
 
 function! utils#cmake#setEnv(name) abort
-    let l:cmake_kit  = get( g:cmake_kits, a:name, {} )
+    let l:cmake_kit = get( s:loaded_cmake_kits, a:name, {} )
     for [key, val] in items( get( l:cmake_kit, 'environment_variables', {} ) )
         execute printf('let $%s="%s"', key, val)
     endfor
 endfunction
 
 function! utils#cmake#unsetEnv(name) abort
-    let l:cmake_kit  = get( g:cmake_kits, a:name, {} )
+    let l:cmake_kit = get( s:loaded_cmake_kits, a:name, {} )
```

Comment:
Why don't you use `g:cmake_kits`?
What happens if I have  changed this variable in the runtime?

Reply:
Because there are multiple ways to supply `cmake_kits` now.

If you have a `.cmake-kits.json`, that file will be read. No matter what you do to `g:cmake_kits`, as long as you have `.cmake-kits.json` or `g:cmake_kits_global_path`, `g:cmake_kits` won't be used. I added that to `README`.

If you have no files and only use `g:cmake_kits`, then it will reload when you run `CMake` or call `SelectKit` or manually run `:call utils#cmake#reloadCMakeKits()`.